### DISCUSSION TITLE
What reading sequence should I follow for learning purpose?

### DISCUSSION THREAD

**Discussion by realzhujunhao** [tikv/raft-rs#514] [Q&A] [answered]
I am currently an undergraduate computer science student, and I hope to learn Raft algorithm by actually looking into implementation. Which ones of source files are the most worth reading pieces to get start?

Any help would be appreciated : )

**Comment by BusyJay** [comment, marked as answer]
I would recommend to read the paper first. And then read the crate docs. And then read the examples and go through all the execution steps.
